src/index.js

```diff
@@ -42,7 +42,7 @@ class UglifyJsPlugin {
       exclude,
       uglifyOptions: {
         output: {
-          comments: false,
+          comments: extractComments ? false : /^\**!|@preserve|@license|@cc_on/,
```

Comment:
Shouldn't this be just the `{RegExp}` to always preserve this comments ? Why is the `extractComments` check needed here ?

Reply:
@michael-ciniawsky When we `extractComments` comments we should remove their from source code. If you pass `extractComments: true` and `comments: /^\**!|@preserve|@license|@cc_on/` output will be contain legal comments in source file and generate `name.js.LICENSE` file

Reply:
Isn't  `extractComments` not only responsible for **only** extracting what's preseverd via `uglifyOptions.output.comments` ?

```js
// Preserve comments based on options, defaults to /^\**!|@preserve|@license|@cc_on/
if (uglifyOptions.output.comments) {
   // Extract all preserved comments
   if (options.extractComments) {
       // [name].js.LICENSE
       extractComments()
   }
   // Leave comments in the source 
}
```

Why is are these options 'decoupled' ? 🙃 

Anyways if that's more complicated atm and this PR now leaves the 'default'  preserved comments in the source aswell as `options.extractComments` also preserves the same default comments (with it's own logic) when extracting them into a separate file, then let's do it :)

Reply:
@michael-ciniawsky `extractComments` does not responsible uglify comments, they are separatly. If think about your logic. But sometimes people leave comment in source code and generate license file to for allow difference type of distribution.